test/iontests.ts

```diff
@@ -45,65 +46,50 @@ function exhaust(reader: Reader) {
+        let event = originalEvents[i];
+        let textEvent = textEvents[i];
+        let binEvent = binEvents[i];
+        //if were equal everybody equals everybody
+        let comparisons : any = [];
+        if (event.annotations[0] === 'embedded_documents') {
+            for (let j = 0; j < event.ionValue.length - 1; j++) {
+                comparisons.push([new IonEventStream(ion.makeReader(event.ionValue[j].ionValue)), new IonEventStream(ion.makeReader(textEvent.ionValue[j].ionValue)), new IonEventStream(ion.makeReader(binEvent.ionValue[j].ionValue))]);
+            }
+        } else {//were in an sexp
+            for (let j = 0; j < event.ionValue.length - 1; j++) {
+                comparisons.push([event.ionValue[j], textEvent.ionValue[j], binEvent.ionValue[j]]);
+                if (event.ionValue[j].eventType === IonEventType.CONTAINER_START) {
+                    j += event.ionValue[j].ionValue.length
+                }
+            }
         }
 
-        containerIdx += 1;
-        reader.stepOut();
-    }
-    return values;
-}
-
-function bytesToString(c: number, idx: number, bytes: Uint8Array) {
-    let sb = '[' + c + '][' + idx + ']';
-    if (bytes.length >= 4 && bytes[0] === 224 && bytes[1] === 1 && bytes[2] === 0 && bytes[3] === 234) {
-        // Ion binary, convert to hex string
-        bytes.forEach(b => {
-            sb += ' ' + ('0' + (b & 0xFF).toString(16)).slice(-2);
-        });
-        return sb;
-    }
-    // otherwise, text:
-    return sb + ' ' + String.fromCharCode.apply(null, bytes);
-}
-
-function equivsTestCompare(c: number, i: number, j: number, bytesI: Uint8Array, bytesJ: Uint8Array, expectedEquivalence: boolean) {
-    let eventStreamI = new IonEventStream(ion.makeReader(bytesI));
-    let eventStreamJ = new IonEventStream(ion.makeReader(bytesJ));
-    let result = eventStreamI.equals(eventStreamJ);
-    assert(expectedEquivalence ? result : !result,
-        'Expected ' + bytesToString(c, i, bytesI)
-        + ' to' + (expectedEquivalence ? '' : ' not')
-        + ' be equivalent to ' + bytesToString(c, j, bytesJ));
-}
-
-function equivsTest(path: string, expectedEquivalence = true, compare = equivsTestCompare) {
-    let binaryValues = loadValues(path, () => ion.makeBinaryWriter());
-    let textValues = loadValues(path, () => ion.makeTextWriter());
-
-    for (let c = 0; c < binaryValues.length; c++) {
-        for (let i = 0; i < binaryValues[c].length; i++) {
-            for (let j = i + 1; j < binaryValues[c].length; j++) {
-                compare(c, i, j, binaryValues[c][i], binaryValues[c][j], expectedEquivalence);
-                compare(c, i, j, binaryValues[c][i], textValues[c][j], expectedEquivalence);
-                compare(c, i, j, textValues[c][i], textValues[c][j], expectedEquivalence);
+        let width = comparisons[0].length;
+        for (let j = 0; j < comparisons.length - 1; j++) {
```

Comment:
I've never seen 4 nested for loops (and we're still within the `i` loop, so we're actually nested 5 levels deep here).  This deserves some clarifying comments.

And what is `width`, really?

Reply:
and let's have `j` range from `0` to `comparisons.length` -- shouldn't cause any change in behavior, but it's cleaner/more-readable to let the range of `k` govern this (instead of also limiting the range of `j`)

Reply:
Width is just the number of streams were comparing, the "width" of the array of comparisons.